### DISCUSSION TITLE
About difficulty using the ```vercel dev``` command in Nest projects

### DISCUSSION THREAD

**Discussion by mozgbrasil** [vercel/community#7321] [Help] [answered]
### Summary

Hello everything is fine

About difficulty using the ```vercel dev``` command in Nest projects

I'm having trouble using the ```vercel dev``` command in Nest projects

 

### Steps to Reproduce

Hello everything is fine

About difficulty using the ```vercel dev``` command in Nest projects

I'm having trouble using the ```vercel dev``` command in Nest projects

I created a Nest project in 

https://stackblitz.com/?starters=backend

and the following repository was created

https://github.com/mozgbrasil/nestjs-typescript-starter-lastil

In a local environment I did the following procedure

```
cd ~/Downloads

nvm use v20.12.2
npm i -g vercel

git clone https://github.com/mozgbrasil/nestjs-typescript-starter-lastil

cd nestjs-typescript-starter-lastil

```

Then when executing the following command it works as expected

```
npm run start:dev 

```

Then, when executing the following command, "404: NOT_FOUND" is returned

```
vercel dev

```

If I create the following file 

```
tee -a vercel.json << END
{
  "version": 2,
  "builds": [
    {
      "src": "src/main.ts",
      "use": "@vercel/node"
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "src/main.ts",
      "methods": ["GET", "POST", "PUT", "PATCH", "DELETE", "HEAD", "OPTIONS"]
    }
  ]
}
END
```

Then, when executing the following command, "500: INTERNAL_SERVER_ERROR" is returned

```
vercel dev

```

and the following error is displayed in the terminal

```
[Nest] 266602 - 28/06/2024, 21:38:51 LOG [NestApplication] Nest application successfully started +3ms
file:///home/marcio/.nvm/versions/node/v20.12.2/lib/node_modules/vercel/node_modules/@vercel/node/dist/dev-server.mjs:1072
    return listener(req, res);
           ^
TypeError: listener is not a function
    at Server.<anonymous> (file:///home/marcio/.nvm/versions/node/v20.12.2/lib/node_modules/vercel/node_modules/@vercel/node/dist/dev-server.mjs:1072: 12)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

Can you please help me

![Captura de tela de 2024-06-28 21-31-44](https://github.com/vercel/community/assets/17202780/33d2a0f9-e010-4029-b7b2-ed9b86d02a1e)
![Captura de tela de 2024-06-28 21-31-52](https://github.com/vercel/community/assets/17202780/f40dda0a-e6b1-4a06-ad0c-a384ee71a06a)
![Captura de tela de 2024-06-28 21-38-46](https://github.com/vercel/community/assets/17202780/9c6f7f11-4c35-4fb9-bc20-05c810654864)
![Captura de tela de 2024-06-28 21-38-55](https://github.com/vercel/community/assets/17202780/0e9508a6-291d-40a1-b654-81395bdd03e3)
![Captura de tela de 2024-06-28 21-38-58](https://github.com/vercel/community/assets/17202780/7db143c5-4c99-483c-be51-6d263b75ce29)

**Comment by pawlean** [comment, marked as answer]
Hi @mozgbrasil!

I've not been able to replicate this error. However, I'd like to share [a discussion from community members](https://github.com/orgs/vercel/discussions/2323#discussioncomment-6491681) that may be helpful. 

Let me know if that helps!

**Reply by mozgbrasil** [reply]
Hello, good morning, how are you, I just made a correction to my report, thank you I will try

**Comment by 0dvFree** [comment]
Hi I had the same problem, have you solved it yet?
